### ISSUE TITLE
Some simple hyperbolic functions can't be solved

### ISSUE THREAD

**Issue by proy87** [sympy/sympy#17955]
`solveset(sinh(x),x)`

**Comment by oscarbenjamin** [comment]
This also gives a ConditionSet with the wrong domain:
```
In [3]: solveset(sinh(x),x)                                                                                                       
Out[3]: 
⎧            ⎛ 2⋅x    ⎞  -x    ⎫
⎪            ⎝ℯ    - 1⎠⋅ℯ      ⎪
⎨x | x ∊ ℝ ∧ ────────────── = 0⎬
⎪                  2           ⎪
⎩                              ⎭
```
It should be Complexes not Reals.

**Comment by ponpiste** [comment]
We are a group of 4 Carnegie Mellon University students and would like to solve this issue in the context of a class homework.

**Comment by oscarbenjamin** [comment]
That's great. Do you know how to go about fixing it?

I recommend first trying to fix it so that the `ConditionSet` has Complexes as the base set unless the user solves over the reals e.g.:
```
solveset(sinh(x), x, Reals) # Should have Reals as base set
solveset(Sinh(x), x, Complexes) # Should have Complexes as base set
```

Once you've got that then you could look at how to return a useful solution if the base set is Reals.

**Comment by ponpiste** [comment]
Yes, we solved it. The problem was that the solve_trig is called for hyperbolic functions. What we did is rewrite the hyperbolic function into exponential function before solving it as a non-trig function. Now the results are correct and simplified. If we make a pull request, can you accept it? That way we receive extra credit.

**Comment by ponpiste** [comment]
<img width="428" alt="Screen Shot 2019-12-02 at 7 06 54 PM" src="https://user-images.githubusercontent.com/34235547/70005263-edc8d280-1536-11ea-9d9b-970d691f6bda.png">

**Comment by oscarbenjamin** [comment]
That looks great! Please do send a PR.

**Comment by ponpiste** [comment]
We will add testcases and documentation as well that cover this very issue. After that we will submit a PR.

**Comment by oscarbenjamin** [comment]
Thanks for this. You can also send a PR with your work so far and then add to after by pushing more commits.

**Comment by gschintgen** [comment]
Hi, isn't this issue a duplicate of #9531?
There's also a pull request #14053 that I tried reviving, but my changes did not get any reviews. Unfortunately I no longer have much time to spare, but maybe some of the changes or additional tests from that PR could prove useful here. So this is just a quick heads up.
It'd be nice to see this fixed. Thanks!

**Comment by oscarbenjamin** [comment]
@gschintgen yes you're right. I hadn't seen those. It looks like #14053 is a more general fix.

**Comment by gschintgen** [comment]
With current master:
```
In [1]: solveset(sinh(x),x)
Out[1]: {2⋅n⋅ⅈ⋅π | n ∊ ℤ} ∪ {ⅈ⋅(2⋅n⋅π + π) | n ∊ ℤ}
```
This has been fixed by #14053.
@ponpiste: thanks for your efforts!
